# SPDX-FileCopyrightText: 2024 Daniel Sampliner <samplinerD@gmail.com>
#
# SPDX-License-Identifier: AGPL-3.0-or-later

diff --git a/build.gradle b/build.gradle
index 710cad2..0c8af11 100644
--- a/build.gradle
+++ b/build.gradle
@@ -1,6 +1,6 @@
 buildscript {
 	dependencies {
-		classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.10'
+		classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.9.25'
 	}
 }
 
@@ -17,14 +17,14 @@ dependencies {
 	testImplementation 'junit:junit:4.13'
 	testImplementation 'com.google.truth:truth:1.0.1'
 
-	r8 'com.android.tools:r8:2.0.99'
+	r8 'com.android.tools:r8:8.5.35'
 }
 
 def fatJarProvider = tasks.register('fatJar', Jar) { task ->
 	task.dependsOn(configurations.named('runtimeClasspath'))
 	task.dependsOn(tasks.named('jar'))
 
-	task.classifier 'fat'
+	task.archiveClassifier = 'fat'
 
 	task.manifest {
 		attributes 'Main-Class': 'com.jakewharton.asciinema.vsync.-Main'
@@ -44,7 +44,7 @@ def fatJarProvider = tasks.register('fatJar', Jar) { task ->
 	}
 }
 
-def r8File = new File("$buildDir/libs/$archivesBaseName-r8.jar")
+def r8File = new File("$buildDir/libs/${base.archivesName.get()}-r8.jar")
 def r8RulesFile = file('src/main/rules.txt')
 def r8Jar = tasks.register('r8Jar', JavaExec) { task ->
 	task.dependsOn(configurations.named('runtimeClasspath'))
@@ -53,7 +53,7 @@ def r8Jar = tasks.register('r8Jar', JavaExec) { task ->
 	task.outputs.file(r8File)
 
 	task.classpath(configurations.r8)
-	task.main = 'com.android.tools.r8.R8'
+	task.mainClass = 'com.android.tools.r8.R8'
 	task.args = [
 		'--release',
 		'--classfile',
@@ -62,11 +62,11 @@ def r8Jar = tasks.register('r8Jar', JavaExec) { task ->
 		'--lib', System.properties['java.home'].toString()
 	]
 	doFirst {
-		task.args += fatJarProvider.get().archivePath
+		task.args += fatJarProvider.get().archiveFile.get()
 	}
 }
 
-def binaryFile = new File("$buildDir/${archivesBaseName}.jar")
+def binaryFile = new File("$buildDir/${base.archivesName.get()}.jar")
 def binaryJar = tasks.register('binaryJar') { task ->
 	task.dependsOn(r8Jar)
 
